### PR TITLE
Address large header-anchor SVGs in RSS feed.

### PR DESCRIPTION
**Pull request by jgerigmeyer**
This PR:

- strips permalinks from markdown headers in RSS feed only
- adds height/width attributes to all svgs

> Hello hello!
>
> First and foremost, thank you for all of the good things you do for the web; I read all of it and I've learned quite a bit over the years.
>
> I use NetNewsWire.app and when I come across an Oddbird post, the link icons look like this (attached). I'm not sure if this is better fixed by the folks at NetNewsWire, or if it's something that could be adjusted on your end — perhaps that SVG needs a height and width defined? Either way, no biggy, but I thought I'd mention it in case others see it as well.

Screenshot 2024-03-26 at 6 58 31 PM

### CODE REVIEW

**Comment by jgerigmeyer** [comment]
@mirisuzanne This is ready for review. I don't know a lot about SVGs, so let me know if any of the height/widths I added don't make sense.

File: src/scss/initial/_headings.scss [hunk 1]
```diff
@@ -37,7 +37,7 @@ h3 {
       var(--heading-start),
       var(--heading-end)
     );
-    -webkit-background-clip: text;
+    background-clip: text;
     background-clip: text;
```
Comment (mirisuzanne) [inline src/scss/initial/_headings.scss:41]: This seems like a mistake? Did you update a linter?

File: src/scss/layout/_main.scss [hunk 2]
```diff
@@ -22,7 +22,7 @@
     grid-column: full;
   }
 
-  @media (min-width: 45ch) {
+  @media (width >= 45ch) {
```
Comment (mirisuzanne) [inline src/scss/layout/_main.scss:25]: Similar, this is maybe an ok change - but it reduces backwards compat. Is a linter doing it, or is there a reason we want to?

**Review by mirisuzanne — COMMENTED** [review]
Changes to SVGs look good. Less convinced by the changes to CSS. I didn't have a chance to do a full test with an RSS reader yet…

File: src/scss/layout/_main.scss (hunk 2 above)
Comment (jgerigmeyer) [reply src/scss/layout/_main.scss]: Yeah, this seems to be automatically changed by Stylelint: https://stylelint.io/user-guide/rules/media-feature-range-notation/

File: src/scss/initial/_headings.scss (hunk 1 above)
Comment (jgerigmeyer) [reply src/scss/initial/_headings.scss]: Hrm, yeah -- this was changed by Stylelint: https://stylelint.io/user-guide/rules/property-no-vendor-prefix/

**Comment by jgerigmeyer** [comment]
> Changes to SVGs look good. Less convinced by the changes to CSS. I didn't have a chance to do a full test with an RSS reader yet…

@mirisuzanne Yeah, good catch. I disabled those Stylelint changes for now, but let me know or feel free to disable or alter the rules globally.

File: src/filters/type.js [hunk 3]
```diff
@@ -108,6 +115,49 @@ export const elide = (html, count = 50) => {
...
+        'link',
+        'listing',
+        'meta',
+        'noscript',
+        'object',
+        'plaintext',
+        'script',
+        'xmp',
+      ].includes(node.tag);
+      const isCustom = ['is-land', 'lite-youtube'].includes(node.tag);
```
Comment (jgerigmeyer) [inline src/filters/type.js:148]: @mirisuzanne After running this through the [W3C validator](https://validator.w3.org/feed/check.cgi?url=https://deploy-preview-550--oddleventy.netlify.app/feed.atom#l799), I figured we probably want to strip out these unsafe and custom tags too. I made the change in https://github.com/oddbird/oddleventy/pull/550/commits/82890cb3d63379ae0cd91e79328957e62ee068d8 -- what do you think?

Comment (mirisuzanne) [reply src/filters/type.js:148]: That seems fine to me. I guess check the build output to see if anything surprising is missing?

Comment (jgerigmeyer) [reply src/filters/type.js:148]: Out of `feed.atom` looks fine to me -- just with anchor links, a few <script> tags, and custom elements removed.

@mirisuzanne I think this is ready to merge once you've given final review?

Comment (jgerigmeyer) [reply src/filters/type.js:148]: @mirisuzanne Do you think this is good to merge?

Comment (mirisuzanne) [reply src/filters/type.js:148]: yep, lgtm

**Review by mirisuzanne — APPROVED** [review]

**Pull request merged**